### PR TITLE
tools, edbg: fix compiler issue on macos

### LINKED ISSUE #7200: edbg: hidapi issue on OS X
I only took a few weeks off... Before then, `make flash` to my samd21-xpro board used my brew installed version openocd. Life was good.

I grabbed latest riot master an hour ago, and `make flash` now tries to fetch and build `edbg`. Unfortunately it crashes and burns:

```
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-recursive
Making all in mac
  CC       hid.lo
hid.c:50:90: warning: unused parameter 'attr' [-Wunused-parameter]
static int pthread_barrier_init(pthread_barrier_t *barrier, const pthread_barrierattr_t *attr, unsigned int count)
                                                                                         ^
hid.c:252:20: warning: comparison of integers of different signs: 'CFIndex' (aka 'long') and 'size_t' (aka 'unsigned long')
      [-Wsign-compare]
                if (chars_copied == len)
                    ~~~~~~~~~~~~ ^  ~~~
hid.c:543:65: warning: unused parameter 'result' [-Wunused-parameter]
static void hid_device_removal_callback(void *context, IOReturn result,
                                                                ^
hid.c:544:47: warning: unused parameter 'sender' [-Wunused-parameter]
                                        void *sender)
                                              ^
hid.c:556:57: warning: unused parameter 'result' [-Wunused-parameter]
static void hid_report_callback(void *context, IOReturn result, void *sender,
                                                        ^
hid.c:556:71: warning: unused parameter 'sender' [-Wunused-parameter]
static void hid_report_callback(void *context, IOReturn result, void *sender,
                                                                      ^
hid.c:557:42: warning: unused parameter 'report_type' [-Wunused-parameter]
                         IOHIDReportType report_type, uint32_t report_id,
                                         ^
hid.c:557:64: warning: unused parameter 'report_id' [-Wunused-parameter]
                         IOHIDReportType report_type, uint32_t report_id,
                                                               ^
hid.c:1023:60: warning: unused parameter 'dev' [-Wunused-parameter]
int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
                                                           ^
hid.c:1023:69: warning: unused parameter 'string_index' [-Wunused-parameter]
int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
                                                                    ^
hid.c:1023:92: warning: unused parameter 'string' [-Wunused-parameter]
int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
                                                                                           ^
hid.c:1023:107: warning: unused parameter 'maxlen' [-Wunused-parameter]
int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
                                                                                                          ^
hid.c:1031:68: warning: unused parameter 'dev' [-Wunused-parameter]
HID_API_EXPORT const wchar_t * HID_API_CALL  hid_error(hid_device *dev)
                                                                   ^
13 warnings generated.
  CCLD     libhidapi.la
clang: warning: argument unused during compilation: '-pthread' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-pthread' [-Wunused-command-line-argument]
Making all in hidtest
  CXX      hidtest.o
  CXXLD    hidtest
arm-none-eabi-g++: error: CoreFoundation: No such file or directory
arm-none-eabi-g++: error: IOKit: No such file or directory
arm-none-eabi-g++: error: unrecognized command line option '-framework'
arm-none-eabi-g++: error: unrecognized command line option '-framework'
arm-none-eabi-g++: error: unrecognized command line option '-pthread'
make[5]: *** [hidtest] Error 1
make[4]: *** [all-recursive] Error 1
make[3]: *** [all] Error 2
make[2]: *** [hidapi/mac/.libs/libhidapi.a] Error 2
make[1]: *** [all] Error 2
make: *** [/Users/travisg/Projects/riotRIOT/dist/tools/edbg/edbg] Error 2
```

I have no idea why this is. And am only slightly cranky that it has regressed. :/

### PR DESCRIPTION
**Pull request by smlng**
fixes #7200 

simply unsetting the compiler vars (CC, CXX) helps, at least for me.

### CODE REVIEW

**Comment by miri64** [comment]
Confirming that with this patch it still compiles under Linux.

**Comment by smlng** [comment]
@miri64 good point, thanks for verifying!

**Comment by emmanuelsearch** [comment]
I'm trying this on my Mac and I get the error:

```
/RIOT/dist/tools/edbg/edbg  -t atmel_cm0p -b -e -v -p -f /RIOT/examples/gnrc_networking/bin/samr21-xpro/gnrc_networking.bin
bash: /RIOT/dist/tools/edbg/edbg: cannot execute binary file
make: *** [flash] Error 126
```

**Comment by smlng** [comment]
@emmanuelsearch looks like a different problem, something with wrong with access rights? But at least `edgb` was compiled which is this PR about.

**Comment by emmanuelsearch** [comment]
not sure what you mean, but if I remove the edbg directory in dist/tools it does not build

```
   text	   data	    bss	    dec	    hex	filename
  80860	    196	  20732	 101788	  18d9c	/Users/emmanuel/Documents/work/CODING/RIOT/examples/gnrc_networking/bin/samr21-xpro/gnrc_networking.elf
[INFO] edbg binary not found - building it from source now
CC= CFLAGS= make -C /Users/emmanuel/Documents/work/CODING/RIOT/dist/tools/edbg
make: *** /Users/emmanuel/Documents/work/CODING/RIOT/dist/tools/edbg: No such file or directory.  Stop.
make: *** [/Users/emmanuel/Documents/work/CODING/RIOT/dist/tools/edbg/edbg] Error 2
```

**Comment by kYc0o** [comment]
I know that is not related to the PR in itself, but `$(RIOTBASE)` is not defined in the edbg Makefile, which doesn't allow to "interact" with it standalone. I think this is one of the "benefits" of having external tools, since they can be compiled and used separately.

Otherwise, it works on my side.

**Comment by kYc0o** [comment]
@emmanuelsearch can you show us the output of `CC --version` and `cc --version` ? It seems that CC is maybe not defined in your setup.

**Comment by thomaseichinger** [comment]
@emmanuelsearch The message in [your comment](https://github.com/RIOT-OS/RIOT/pull/7283#issuecomment-312260283) makes me wonder if you deleted the whole edbg directory and thus the Makefile in `/Users/emmanuel/Documents/work/CODING/RIOT/dist/tools/edbg`?

File: dist/tools/edbg/Makefile [hunk 1]
```diff
@@ -4,6 +4,9 @@ PKG_VERSION=935bb4b1efd0e2f8434cf97443e0d95b9e02ec1d
 PKG_LICENSE=BSD-3-Clause
 PKG_BUILDDIR=$(CURDIR)/bin
 
+CC=
```
Comment (thomaseichinger) [inline dist/tools/edbg/Makefile]: Can we have a comment explaining the rational for this?

**Comment by thomaseichinger** [comment]
@travisgriggs Would you mind trying if this change solves #7200 for you too?

**Comment by emmanuelsearch** [comment]
@thomaseichinger I thought I had tried this variant too (leaving only the makefile) but maybe I had made a mistake.
Anyways, now what I get is the following warnings:

```
make -Chidapi
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-recursive
Making all in mac
  CC       hid.lo
hid.c:50:90: warning: unused parameter 'attr' [-Wunused-parameter]
static int pthread_barrier_init(pthread_barrier_t *barrier, const pthread_barrierattr_t *attr, unsigned int count)
                                                                                         ^
hid.c:252:20: warning: comparison of integers of different signs: 'CFIndex' (aka 'long') and 'size_t' (aka 'unsigned long')
      [-Wsign-compare]
                if (chars_copied == len)
                    ~~~~~~~~~~~~ ^  ~~~
hid.c:543:65: warning: unused parameter 'result' [-Wunused-parameter]
static void hid_device_removal_callback(void *context, IOReturn result,
                                                                ^
hid.c:544:47: warning: unused parameter 'sender' [-Wunused-parameter]
                                        void *sender)
                                              ^
hid.c:556:57: warning: unused parameter 'result' [-Wunused-parameter]
static void hid_report_callback(void *context, IOReturn result, void *sender,
                                                        ^
hid.c:556:71: warning: unused parameter 'sender' [-Wunused-parameter]
static void hid_report_callback(void *context, IOReturn result, void *sender,
                                                                      ^
hid.c:557:42: warning: unused parameter 'report_type' [-Wunused-parameter]
                         IOHIDReportType report_type, uint32_t report_id,
                                         ^
hid.c:557:64: warning: unused parameter 'report_id' [-Wunused-parameter]
                         IOHIDReportType report_type, uint32_t report_id,
                                                               ^
hid.c:1023:60: warning: unused parameter 'dev' [-Wunused-parameter]
int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
                                                           ^
hid.c:1023:69: warning: unused parameter 'string_index' [-Wunused-parameter]
int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
                                                                    ^
hid.c:1023:92: warning: unused parameter 'string' [-Wunused-parameter]
int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
                                                                                           ^
hid.c:1023:107: warning: unused parameter 'maxlen' [-Wunused-parameter]
int HID_API_EXPORT_CALL hid_get_indexed_string(hid_device *dev, int string_index, wchar_t *string, size_t maxlen)
                                                                                                          ^
hid.c:1031:68: warning: unused parameter 'dev' [-Wunused-parameter]
HID_API_EXPORT const wchar_t * HID_API_CALL  hid_error(hid_device *dev)
                                                                   ^
13 warnings generated.
  CCLD     libhidapi.la
clang: warning: argument unused during compilation: '-pthread' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-pthread' [-Wunused-command-line-argument]
Making all in hidtest
  CXX      hidtest.o
  CXXLD    hidtest
clang: warning: argument unused during compilation: '-pthread' [-Wunused-command-line-argument]
make[5]: Nothing to be done for `all-am'.
gcc  -Ihidapi/hidapi -W -Wall -Wextra -O2 -std=gnu99 dap.c edbg.c target.c target_atmel_cm0p.c target_atmel_cm3.c target_atmel_cm4.c target_atmel_cm7.c dbg_mac.c hidapi/mac/.libs/libhidapi.a -framework IOKit -framework CoreFoundation -o edbg
target_atmel_cm0p.c:86:7: warning: missing field 'name' initializer [-Wmissing-field-initializers]
  { 0 },
      ^
1 warning generated.
target_atmel_cm3.c:94:7: warning: missing field 'name' initializer [-Wmissing-field-initializers]
  { 0 },
      ^
1 warning generated.
target_atmel_cm4.c:130:7: warning: missing field 'chip_exid' initializer [-Wmissing-field-initializers]
  { 0 },
      ^
1 warning generated.
mv /Users/emmanuel/Documents/work/CODING/RIOT/dist/tools/edbg/bin/edbg .
[INFO] edbg binary successfully build!
/Users/emmanuel/Documents/work/CODING/RIOT/dist/tools/edbg/edbg  -t atmel_cm0p -b -e -v -p -f /Users/emmanuel/Documents/work/CODING/RIOT/examples/gnrc_networking/bin/samr21-xpro/gnrc_networking.bin
Error: no debuggers found
make: *** [flash] Error 1
```

**Comment by kYc0o** [comment]
The warnings are "Ok" but it seems you don't have  samr21 connected. Do you?

File: dist/tools/edbg/Makefile (hunk 1 above)
Comment (smlng) [inline dist/tools/edbg/Makefile]: RIOT sets the CC variables when evaluation the toolchain, and for `samr21-xpro` its `gcc-arm-none-eabi` which runs into an error on macOS, see #7200 error message:

```
arm-none-eabi-g++: error: CoreFoundation: No such file or directory
arm-none-eabi-g++: error: IOKit: No such file or directory
arm-none-eabi-g++: error: unrecognized command line option '-framework'
arm-none-eabi-g++: error: unrecognized command line option '-framework'
arm-none-eabi-g++: error: unrecognized command line option '-pthread'
```

As `edbg` is build for flashing, that is **after** RIOT was already build, its safe to simply unset `CC` and let `edbg`s own config+make find the correct compiler again.

**Comment by smlng** [comment]
@emmanuelsearch: the warnings are to be ignored, as @kYc0o pointed out.

Btw. `edbg` was build according to your output above:

> [INFO] edbg binary successfully build!

and you need to plug in a board ( @kYc0o right again) otherwise you get:

>Error: no debuggers found

----

to conclude (TL;DR) this PR seem to work for you, too

File: dist/tools/edbg/Makefile (hunk 1 above)
Comment (thomaseichinger) [inline dist/tools/edbg/Makefile]: @smlng I know, I was thinking about something like
```make
# CC and CXX variables are set during RIOT's build process
# thus we have to reset them to successfully build on macOS systems
CC=
CXX=
```

Comment (smlng) [inline dist/tools/edbg/Makefile]: ah, good point - will add comments!

**Comment by smlng** [comment]
@thomaseichinger added comment, directly amended.

**Comment by emmanuelsearch** [comment]
actually I had a board connected... but through a faulty USB hub. My bad.
Works for me.

**Comment by aabadie** [comment]
@thomaseichinger just missing an ACK from you here.

After, it will need to be backported (since this is a bug fix IMO).

**Comment by smlng** [comment]
let's give @thomaseichinger some more time, in the meantime I'll open a backport PR.

**Comment by aabadie** [comment]
@thomaseichinger, can you ACK so we can merge ?

**Comment by thomaseichinger** [comment]
ACK & go

**Comment by aabadie** [comment]
Thanks @thomaseichinger.

This one was already backported. So nothing more to do here.

**Pull request merged**